### PR TITLE
[ctx_prof] Fix the pre-thinlink "use" case

### PR DESCRIPTION
**Pull request by mtrofin**
Didn't notice in #101338 that the instrumentation in `llvm/test/Transforms/PGOProfile/ctx-prof-use-prelink.ll` was actually incorrect.

### CODE REVIEW

File: llvm/lib/Passes/PassBuilderPipelines.cpp [hunk 1]
```diff
@@ -1173,13 +1173,12 @@ PassBuilder::buildModuleSimplificationPipeline(OptimizationLevel Level,
   const bool IsMemprofUse = IsPGOPreLink && !PGOOpt->MemoryProfile.empty();
   // We don't want to mix pgo ctx gen and pgo gen; we also don't currently
   // enable ctx profiling from the frontend.
-  assert(
-      !(IsPGOInstrGen && PGOCtxProfLoweringPass::isContextualIRPGOEnabled()) &&
-      "Enabling both instrumented FDO and contextual instrumentation is not "
-      "supported.");
+  assert(!(IsPGOInstrGen && PGOCtxProfLoweringPass::isCtxIRPGOInstrEnabled()) &&
+         "Enabling both instrumented FDO and contextual instrumentation is not "
```
Comment (boomanaiden154) [inline llvm/lib/Passes/PassBuilderPipelines.cpp:1177]: I know this is copy and paste, but `s/FDO/PGO` given everything in the context/variable naming is also PGO?

**Review by boomanaiden154 — APPROVED** [review]
One nit, otherwise LGTM.

File: llvm/test/Transforms/PGOProfile/ctx-prof-use-prelink.ll [hunk 2]
```diff
@@ -7,19 +7,22 @@
 
 declare void @bar()
 
+;.
+; CHECK: @__profn_foo = private constant [3 x i8] c"foo"
+;.
 define void @foo(i32 %a, ptr %fct) {
 ; CHECK-LABEL: define void @foo(
 ; CHECK-SAME: i32 [[A:%.*]], ptr [[FCT:%.*]]) local_unnamed_addr {
+; CHECK-NEXT:    call void @llvm.instrprof.increment(ptr @__profn_foo, i64 728453322856651412, i32 2, i32 0)
```
Comment (snehasish) [inline llvm/test/Transforms/PGOProfile/ctx-prof-use-prelink.ll:16]: Why did this get hoisted out from the `no:` block? The only functional change I could find in this patch was `addLTOPrelinkPasses(MPM)` which adds `CanonicalizeAliasesPass` and `NameAnonGlobalPass`.

Comment (mtrofin) [reply llvm/test/Transforms/PGOProfile/ctx-prof-use-prelink.ll:16]: it's the changes in PGOInstrumentation that do that.

**Review by snehasish — APPROVED** [review]
lgtm

File: llvm/lib/Passes/PassBuilderPipelines.cpp (hunk 1 above)
Comment (mtrofin) [reply llvm/lib/Passes/PassBuilderPipelines.cpp]: done.

**Comment by klausler** [comment]
This patch broke at least one flang build bot (https://lab.llvm.org/buildbot/#/builders/89/builds/3722); please repair or revert with dispatch.

**Pull request merged**